src/config.rs

```diff
@@ -0,0 +1,136 @@
+//! Provides a simple way of reading configuration files.
+//!
+//! Reads a TOML config file for the IPC Agent and deserializes it in a type-safe way into a
+//! [`Config`] struct. Usage:
+//!
+//! ```
+//! let c: Config = Config::from_file("config/example.toml");
+//! ```
+
+use std::collections::HashMap;
+use std::fmt::Formatter;
+use std::fs;
+use std::str::FromStr;
+
+use anyhow::Result;
+use fvm_shared::address::Address;
+use ipc_sdk::subnet_id::SubnetID;
+use serde::de::{Error, SeqAccess};
+use serde::{Deserialize, Deserializer};
+use url::Url;
+
+/// The top-level struct representing the config. Calls to [`Config::from_file`] deserialize into this
+/// struct.
+#[derive(Deserialize)]
+pub(crate) struct Config {
+    pub subnets: HashMap<String, Subnet>,
```

Comment:
The key to the hashmap, is it a name? or it will only be `child` and `root` or it's actually the caller can call whatever it is?

What happens if the `subnets` had different key, but the Subnet values configurations is the same? Do we consider them to the be same or should we raise a warning or let caller be responsible?

Reply:
The key to the hashmap comes from the configuration. If we have two subnets in the config like this:
```
[subnets]
[subnets.root]
# root subnet config
[subnets.child]
# child subnet config
```
Then the hashmap will have keys "root" and "child".